seed/ts-sdk/nullable/src/api/resources/nullable/client/Client.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * This file was auto-generated by Fern from our API Definition.
+ */
+
+import * as core from "../../../../core";
+import * as SeedNullable from "../../../index";
+import urlJoin from "url-join";
+import * as serializers from "../../../../serialization/index";
+import * as errors from "../../../../errors/index";
+
+export declare namespace Nullable {
+    export interface Options {
+        environment: core.Supplier<string>;
+        /** Specify a custom URL to connect the client to. */
+        baseUrl?: core.Supplier<string>;
+    }
+
+    export interface RequestOptions {
+        /** The maximum time to wait for a response in seconds. */
+        timeoutInSeconds?: number;
+        /** The number of times to retry the request. Defaults to 2. */
+        maxRetries?: number;
+        /** A hook to abort the request. */
+        abortSignal?: AbortSignal;
+        /** Additional headers to include in the request. */
+        headers?: Record<string, string>;
+    }
+}
+
+export class Nullable {
+    constructor(protected readonly _options: Nullable.Options) {}
+
+    /**
+     * @param {SeedNullable.GetUsersRequest} request
+     * @param {Nullable.RequestOptions} requestOptions - Request-specific configuration.
+     *
+     * @example
+     *     await client.nullable.getUsers({
+     *         usernames: "usernames",
+     *         avatar: "avatar",
+     *         activated: true,
+     *         tags: null
+     *     })
+     */
+    public async getUsers(
+        request: SeedNullable.GetUsersRequest = {},
+        requestOptions?: Nullable.RequestOptions,
+    ): Promise<SeedNullable.User[]> {
+        const { usernames, avatar, activated, tags } = request;
+        const _queryParams: Record<string, string | string[] | object | object[] | null> = {};
+        if (usernames != null) {
+            if (Array.isArray(usernames)) {
+                _queryParams["usernames"] = usernames.map((item) => item);
```

Comment:
unrelated, but are we creating a copy of an array here? I think there are simpler ways to do this.

Reply:
Yea it looks like it - I have a feeling this is because we always map over the array values and (optionally) perform a `string` mapper function (e.g. `item.toString()`), so it's simpler to _always_ use the `.map` expression.

We can take a closer look here though.